addons/hr_holidays/models/hr_leave.py

```diff
@@ -82,25 +83,36 @@ def default_get(self, fields_list):
             lt = self.env['hr.leave.type'].browse(defaults.get('holiday_status_id'))
             defaults['state'] = 'confirm' if lt and lt.leave_validation_type != 'no_validation' else 'draft'
 
+        now = fields.Datetime.now()
+        if 'date_from' not in defaults:
+            defaults.update({'date_from': now})
+        if 'date_to' not in defaults:
+            defaults.update({'date_to': now})
```

Comment:
Avoid `update()` when possible, it's faster to directly assign :-)

```suggestion
            defaults['date_from'] = now
        if 'date_to' not in defaults:
            defaults['date_to'] = now
```